src/Schedule.php

```diff
@@ -48,4 +51,35 @@ public function materialize(\DateTimeImmutable $from, \DateInterval $period): ar
         }
         return $include;
     }
+
+    /**
+     * @return array
+     */
+    public function getData(): array
+    {
+        $rules = [];
+        foreach ($this->rules as $rule) {
+            $object = new \stdClass();
+            $object->type = get_class($rule);
+            $object->data = json_encode($rule);
+            $rules[] = $object;
+        }
+        return $rules;
+    }
+
+    /**
+     * @return int
+     */
+    public function getId(): int
+    {
+        return $this->id;
+    }
+
+    /**
+     * @param int $id
+     */
+    public function setId(int $id): void
+    {
+        $this->id = $id;
+    }
```

Comment:
If someone taught you that for each private variable you need a getter and setter, don't listen that bad teacher anymore.

1. Every public method should come with a reason
2. If you need to mutate the state of an object, better to name that method somehow connected with the logic of event, like you inform an object about something. Example: instead of `Schedule::setId($id)` you may use `Schedule::persists($id)`.

Reply:
Please read https://www.yegor256.com/2014/09/16/getters-and-setters-are-evil.html

Reply:
Very nice arcticle.
I have read this and have understand what when we talk about objects what can be presented in a real world, getters/setters are not possible to use, but a lot of conversations about objects not implemented in a real, like fixed time subscription rules. Is it ok use getters/setters sometimes when we can't compare our function with real process? 

And i liked possition about 'accessors': “Don’t ask for the information you need to do the work; ask the object that has the information to do the work for you.” I will use it anyway.

Reply:
We can discuss this in person.